src/types/markers.js

```diff
@@ -608,18 +608,30 @@ type VsyncTimestampPayload = {|
   type: 'VsyncTimestamp',
 |};
 
-export type ScreenshotPayload = {|
-  type: 'CompositorScreenshot',
-  // This field represents the data url of the image. It is saved in the string table.
-  url: IndexIntoStringTable,
-  // A memory address that can uniquely identify a window. It has no meaning other than
-  // a way to identify a window.
-  windowID: string,
-  // The original dimensions of the window that was captured. The actual image that is
-  // stored in the string table will be scaled down from the original size.
-  windowWidth: number,
-  windowHeight: number,
-|};
+export type ScreenshotPayload =
+  | {|
+      type: 'CompositorScreenshot',
+      // This field represents the data url of the image. It is saved in the string table.
+      url: IndexIntoStringTable,
+      // A memory address that can uniquely identify a window. It has no meaning other than
+      // a way to identify a window.
+      windowID: string,
+      // The original dimensions of the window that was captured. The actual image that is
+      // stored in the string table will be scaled down from the original size.
+      windowWidth: number,
+      windowHeight: number,
+    |}
+  // Markers that represent the closing of a window (name === 'CompositorScreenshotWindowDestroyed')
+  // only have a windowID data.
+  | {|
+      type: 'CompositorScreenshot',
```

Comment:
I started to feel like we should really give this another type instead of `CompositorScreenshot`. But this will require more changes, we can think about doing it and adding an upgrader later.

Reply:
Yeah, probably the type should be the same as its name, that would make it easier in some parts. I believe Florian chose this path first because then all already existing switch with marker types wouldn't need to be change... but now I think (like you) that this wasn't the best choice.

Reply:
From what I remember, the reason why they shared the same type is that at the time we were automatically hiding the screenshot markers (and showing them only in the screenshot tracks), and I wanted the CompositorScreenshotWindowDestroyed markers to be hidden too without complicating the logic that was hiding them. Now that we are no longer hiding these markers, I think changing the type would make sense.